yb-voyager/cmd/importSchema.go

```diff
@@ -86,24 +86,42 @@ func importSchema() error {
 
 	tconf.Schema = strings.ToLower(tconf.Schema)
 
+	tconfSchema := tconf.Schema
+	// setting the tconf schema to public here for initalisation to handle cases where non-public target schema
+	// is not created as it will be created with `createTargetSchemas` func, so not a problem in using public as it will be
+	// available always and this is just for initialisation of tdb and marking it nil again back.
+	tconf.Schema = "public"
+	tdb = tgtdb.NewTargetDB(&tconf)
+	err = tdb.Init()
+	if err != nil {
+		utils.ErrExit("Failed to initialize the target DB: %s", err)
+	}
+
 	if callhome.SendDiagnostics || getControlPlaneType() == YUGABYTED {
-		tconfSchema := tconf.Schema
-		// setting the tconf schema to public here for initalisation to handle cases where non-public target schema
-		// is not created as it will be created with `createTargetSchemas` func, so not a problem in using public as it will be
-		// available always and this is just for initialisation of tdb and marking it nil again back.
-		tconf.Schema = "public"
-		tdb = tgtdb.NewTargetDB(&tconf)
-		err := tdb.Init()
+		targetDBDetails = tdb.GetCallhomeTargetDBInfo()
+	}
+
+	if tconf.RunGuardrailsChecks {
+		// Check import schema permissions
+		missingPermissions, err := tdb.GetMissingImportSchemaPermissions()
 		if err != nil {
-			utils.ErrExit("Failed to initialize the target DB: %s", err)
+			utils.ErrExit("Failed to get missing import schema permissions: %s", err)
+		}
+		if len(missingPermissions) > 0 {
+			utils.PrintAndLog(color.RedString("The target database is missing the following permissions required for importing schema:"))
+			output := strings.Join(missingPermissions, "\n")
+			utils.PrintAndLog(output)
+			utils.ErrExit("Please grant the required permissions and retry the import.")
```

Comment:
Let's add a prompt here: "Are you sure you want to continue?" instead of exiting directly.

Reply:
From our internal code handling, yes, we can run import-schema without superuser, but I think as we are creating schemas here, which can be various object types, and I am not sure what object types might need a higher privileged user or superuser in that case. So maybe it's better to just ask for it?

Reply:
@makalaaneesh  I think superuser is required for setting some of the session variables in import data.
And since that requires superuser we decided to keep it as it is for import schema also

Reply:
@sanyamsinghal Yes, but check this out - we recently removed this restriction. (we prompt user, and allow them to continue) https://github.com/yugabyte/yb-voyager/pull/1420